extension/k8sleaderelector/go.mod

```diff
@@ -11,6 +11,7 @@ require (
 	go.opentelemetry.io/collector/confmap v1.29.1-0.20250402200755-cb5c3f4fb9dc
 	go.opentelemetry.io/collector/extension v1.29.1-0.20250402200755-cb5c3f4fb9dc
 	go.opentelemetry.io/collector/extension/extensiontest v0.123.1-0.20250402200755-cb5c3f4fb9dc
+	go.opentelemetry.io/collector/pipeline v0.123.1-0.20250402200755-cb5c3f4fb9dc
```

Comment:
consider moving the host and elector test impl to the k8sclusterreceiver to avoid this dependency.

Reply:
This extension can be supported in other receivers/exporters. In such case this fake extension implementation can be leveraged to write unit tests. Also the `fakeHost` and mock implementation of functions belong to `k8sleaderelector` package